### DISCUSSION TITLE
Unable to bring vsphere deployment to work

### DISCUSSION THREAD

**Discussion by cbrassel** [siderolabs/talos#7136] [Q&A] [answered]
Hi,
I'm new to Kubernetes and I want to start somewhere, Talos seem's to be a good entry point, but I'm unable to bring this to work despite the simplicity of the process.
I have follow this procedure : https://www.talos.dev/v1.4/talos-guides/install/virtualized-platforms/vmware/ which seem's appropriated, the vm's are created flawlessly, but bootstrapping the cluster seem's to not work for any reason ..
After entering the bootstrap command I'm able to retrieve the kubeconfig but that's all, the api server seem's to not start at all, here a screenshot of the first frontal plane console 
![talos_err](https://user-images.githubusercontent.com/4759371/234567079-89be90f8-310b-4a71-8d07-b26350655874.jpg)
and the output of the commands : 
![talos_err2](https://user-images.githubusercontent.com/4759371/234568052-40952693-4205-46e9-ab2c-fdf4420f1ee1.jpg)
Can some one point me in a direction ? I don't understand where is the problem.

Regard's
Claude

**Comment by smira** [comment]
It's hard to say anything specific without details, but did the bootstrap go as expected?

What does `talosctl -n <IP> etcd members` show?

Generic guide: https://www.talos.dev/v1.4/advanced/troubleshooting-control-plane/

**Reply by cbrassel** [reply]
the etcd members output seem's to be a little bit disturbing for me (regarding the node) : 
talosctl -n 10.35.1.155 etcd members
NODE          ID                 HOSTNAME        PEER URLS                  CLIENT URLS                LEARNER
10.35.1.155   78ef426f57aeccea   talos-0mv-vf1   https://10.35.1.155:2380   https://10.35.1.155:2379   false
10.35.1.155   a966fdce4f1bf477   talos-fii-k97   https://10.35.1.157:2380   https://10.35.1.157:2379   false
10.35.1.155   e563b9561dc4a761   talos-ket-4q9   https://10.35.1.156:2380   https://10.35.1.156:2379   false

Following the troubleshot guide it appears that : 
service etcd is running
service kubelet is running

but the manifest seem's not appearing : 
talosctl -n 10.35.1.155 ls /etc/kubernetes/manifests
NODE          NAME
10.35.1.155   .

talosctl logs controller-runtime 
..
10.35.1.155: 2023-04-26T13:22:10.190Z ERROR controller failed {"component": "controller-runtime", "controller": "k8s.ManifestApplyController", "error": "error creating mapping for object /v1/Secret/bootstrap-token-szkosa: Get \"https://localhost:6443/api?timeout=32s\": dial tcp [::1]:6443: connect: connection refused"}
..

and no container is running : 
talosctl -n 10.35.1.155 containers --kubernetes
NODE   NAMESPACE   ID   IMAGE   PID   STATUS


Regard's
Claude

**Reply by smira** [reply]
the `/etc/kubernetes/manifests` is outdated, we should update the docs. 

so etcd members is fine, and the static pod manifests are now handled differently. 

you should see `rendered new static pod` messages from Talos

if those appeared, now everything is on the kubelet side: `talosctl logs kubelet`.

**Reply by cbrassel** [reply]
didn't find any `rendered new static pod` in the log, I have attached the log
[kubelet.log](https://github.com/siderolabs/talos/files/11333514/kubelet.log)

**Reply by cbrassel** [reply]
and thank's for your help

**Reply by cbrassel** [reply]
maybe is this the reason ? 
failed to get sandbox image \"registry.k8s.io/pause:3.6\": failed to pull image \"registry.k8s.io/pause:3.6\": failed to pull and unpack image \"registry.k8s.io/pause:3.6\": failed to resolve reference \"registry.k8s.io/pause:3.6\": pulling from host registry.k8s.io failed with status code [manifests 3.6]: 503 Service Unavailable"}

**Reply by smira** [reply]
yes, that is definitely a problem, but the image itself is fine, it's something in your environment

**Reply by cbrassel** [reply]
I don't see what can be wrong, the network has unlimited access to the internet, dns resolution goes over google (juste made the change), default gw is the right one (ntp is working) ..

**Reply by cbrassel** [reply]
I have just tried via docker on my desktop and the cluster doesn't fire up either .. it get stuck after creating the 2 containers .. And the network environment is completely different, so I'm not sure it's because of my environment ..

**Reply by smira** [reply]
I don't have any exact idea, you can try `docker pull` yourself with that image. You can also try restarting the kubelet with `talosctl services kubelet restart`

**Reply by cbrassel** [reply]
It seem's the registry.k8s.io have some trouble .. I have tested from our production VMware cluster, from my desktop and from another server, same 503 .. : 

`docker pull registry.k8s.io/pause:3.6`
        Error response from daemon: received unexpected HTTP status: 503 Service Unavailable
`docker pull registry.k8s.io/kube-apiserver:v1.27.1`
        Error response from daemon: received unexpected HTTP status: 503 Service Unavailable
`docker pull gcr.io/etcd-development/etcd:v3.5.8-arm64`
v3.5.8-arm64: Pulling from etcd-development/etcd
dde16751a2cc: Pull complete
fe5ca62666f0: Pull complete
fff4e558ad3a: Pull complete
fcb6f6d2c998: Pull complete
e8c73c638ae9: Pull complete
1e3d9b7d1452: Pull complete
4aa0ea1413d3: Pull complete
7c881f9ab25e: Pull complete
5627a970d25e: Pull complete
f5bd09c760b4: Pull complete
074cb99a821d: Pull complete
7989df9becdf: Pull complete
4912cb702652: Pull complete
6c820455abdd: Pull complete
Digest: sha256:e0f5d5d4cf8bba477e34a4a0506aec97695dc0e9da2279f98c3f5da31075092d
Status: Downloaded newer image for gcr.io/etcd-development/etcd:v3.5.8-arm64
gcr.io/etcd-development/etcd:v3.5.8-arm64

**Reply by cbrassel** [reply]
just made a test from a host azur host and the docker pull is working as expected ... very strange ! 

I will try from my home ip in a couple of hours .. 

Thank's for the help, I will let you know if I have found the solution ..

Regard's
Claude

**Reply by smira** [reply]
Also keep in mind that `registry.k8s.io` is a CDN, so it might resolve to different services depending on your location. So it might be that some of the parts of the CDN have issues. Unfortunately not much we can help on Talos side. There's a way to override pause image, but I guess it's not a production way.

**Reply by frezbo** [reply, marked as answer]
I think this might be the issue with pulling images https://github.com/kubernetes/registry.k8s.io/issues/234

**Reply by cbrassel** [reply]
since midnight the pull is working again, so the cluster has finished his setup and everything is working now ...
At least I know now how to debug Talos !

Thank's for your time.
